### PR TITLE
Added verbose oneDNN lib version

### PR DESCRIPTION
**Pull request by arlesniak**
### PR types

Others 
### PR changes

Others 
### Describe

Added verbose oneDNN lib version as part of customer request in [issue 27935](https://github.com/PaddlePaddle/Paddle/issues/27935)

### CODE REVIEW

File: paddle/fluid/platform/device_context.h [hunk 1]
```diff
@@ -485,6 +486,7 @@ class MKLDNNDeviceContextThreadLocals {
     void set_cur_input_shape_cache_capacity(int input_shape_cache_capacity);
     void set_cur_paddle_data_layout(framework::DataLayout dl);
     framework::DataLayout get_cur_paddle_data_layout(void);
+    void announce(void);
```
Comment (arogowie-intel) [inline paddle/fluid/platform/device_context.h]: Please add some doc explaining what is this function doing.

**Review by arogowie-intel — COMMENTED** [review]
I'd try to do this the other way. IMHO adding additional variable to ` MKLDNNDeviceContextThreadLocals::Body` only to check whether some lib version information was logged once is not sufficient reason. Moreover with such solution this condition have to be checked at construction of each MKLDNN operator (which uses ~Handler[T] class ). 

I'm not aware what is the main usage and reason to print such information, but
I'd rather convert this to _helper_ function like `printDnnlLibraryVersion` and make sure it's called at place which guarantees it will be called only once. Some Executor constructor or other place?

File: paddle/fluid/platform/device_context.h (hunk 1 above)
Comment (jczaja) [reply paddle/fluid/platform/device_context.h]: @arogowie-intel  Operators can be called directly without supervision of Executor, NaiveExecutor, ParallelExecutor.. There is also dygraph with its own specifc code flow. so I'm not sure if you can make it more elegant solution and faster than this

**Comment by arlesniak** [comment]
@arogowie-intel  Thank you for comment. The root of the PR is the need to tell whether training was actually run on oneDNN by inspecting logs without extra effort.
This could also be done via inspecting env variables or calling some API function from every training script but that way was rejected by customer. Also suppressing the info with VLOG levels would force customer to use extra effort i.e. special env variables to bring the information to the log. 
Having in mind dynamic and static execution modes and possibility of running ops in tests without executor, the `MKLDNNDeviceContextThreadLocals `looks like the right place. `~Handler[T]` points out that OneDNN execution path was chosen and executed.

**Comment by arogowie-intel** [comment]
@arlesniak @jczaja OK, I wasn't aware that we don't need the `~Executor` object. With that information I think my comment is now irrelevant.

File: paddle/fluid/platform/device_context.h [hunk 2]
```diff
@@ -485,6 +486,7 @@ class MKLDNNDeviceContextThreadLocals {
     void set_cur_input_shape_cache_capacity(int input_shape_cache_capacity);
     void set_cur_paddle_data_layout(framework::DataLayout dl);
     framework::DataLayout get_cur_paddle_data_layout(void);
+    void started_info(void);
```
Comment (arogowie-intel) [inline paddle/fluid/platform/device_context.h]: I'd suggest to rename it like `log_lib_version` to name what it exactly does.

**Review by arogowie-intel — DISMISSED** [review]
Just have minor suggestion for better readability. Other than that LGTM.

File: paddle/fluid/platform/device_context.h (hunk 2 above)
Comment (arlesniak) [reply paddle/fluid/platform/device_context.h]: Done

**Comment by arlesniak** [comment]
@luotao1 Could you please start your review?

**Comment by luotao1** [comment]
Could you tell us how to use this verbose and paste the result?

**Comment by arlesniak** [comment]
During running training scripts and tests, whenever oneDNN is to be used, the following info would be printed to stdout (timestamp is exemplary):
`I1210 09:07:51.475006 80876 device_context.cc:473] oneDNN v1.6.0`

It will be printed once per oneDNN lib initial usage in executor/process. 
In a log file it will be printed at moment (line) when the oneDNN will be actually used for the first time.
When oneDNN is enabled, common appearance will be one time per training log without any additional env variables or training script changes.

**Comment by luotao1** [comment]
Please solve the conflict

**Review by luotao1 — APPROVED** [review]
LGTM! I test it in models/dygraph.

**Comment by luotao1** [comment]
Please cherry-pick into release/2.0

**Pull request merged**